ext/php7/ddtrace.c

```diff
@@ -558,12 +548,62 @@ static PHP_RSHUTDOWN_FUNCTION(ddtrace) {
     if (ZSTR_LEN(get_DD_TRACE_REQUEST_INIT_HOOK())) {
         dd_request_init_hook_rshutdown();
     }
+}
 
-    zai_config_rshutdown();
+static PHP_RSHUTDOWN_FUNCTION(ddtrace) {
+    UNUSED(module_number, type);
+
+    if (!get_DD_TRACE_ENABLED()) {
+        return SUCCESS;
+    }
+
+    ddtrace_close_all_open_spans();  // All remaining non-internal userland spans
+    if (DDTRACE_G(open_spans_top) && DDTRACE_G(open_spans_top)->execute_data == NULL) {
+        // we have a root span. Close it.
+        dd_trace_stop_span_time(&DDTRACE_G(open_spans_top)->span);
+        ddtrace_close_span(DDTRACE_G(open_spans_top));
+    }
+    if (ddtrace_flush_tracer() == FAILURE) {
+        ddtrace_log_debug("Unable to flush the tracer");
+    }
+
+    dd_clean_globals();
 
     return SUCCESS;
 }
 
+int ddtrace_post_deactivate(void) {
+    // zai config may be accessed indirectly via other modules RSHUTDOWN, so delay this until the last possible time
```

Comment:
I didn't understand this comment. How is this delaying zai config's shutdown? Following it through to `zai_config_runtime_config_dtor`, it looks like it does shut it down immediately?

Reply:
The point is about it being in post_deactivate vs RSHUTDOWN